packages/core/services/maps-api-loader/lazy-maps-api-loader.ts

```diff
@@ -84,8 +84,9 @@ export class LazyMapsAPILoader extends MapsAPILoader {
   protected _config: LazyMapsAPILoaderConfigLiteral;
   protected _windowRef: WindowRef;
   protected _documentRef: DocumentRef;
+  protected readonly _SCRIPT_ID: string = 'agmGoogleMapsApiScript';
 
-  constructor(@Inject(LAZY_MAPS_API_CONFIG) config: any, w: WindowRef, d: DocumentRef) {
+  constructor(@Optional() @Inject(LAZY_MAPS_API_CONFIG) config: any, w: WindowRef, d: DocumentRef) {
     super();
     this._config = config || {};
```

Comment:
This line could be saved by writing the constructor in ES6 style:
`constructor(@Inject(LAZY_MAPS_API_CONFIG) config: LazyMapsAPILoaderConfigLiteral = {}, w: WindowRef, d: DocumentRef) {`

Reply:
thx, will improve this

Reply:
Actually you'd still need the like `this._config = config`, if you wanna go full-TypeScript mode, it could be:

```
@Optional() @Inject(TOKEN) private _config: Type = {}
```

However, it's not _exactly_ the same. The current way it's written (with `||`) would overwrite a `null` as well. In ES6, only `undefined` (either explicit undefined, which will be the case with DI always, or omitting the parameter completely) will be swapped with a default parameter. Shouldn't be an issue here, though.